FileIO/GmshIO/GMSHPolygonTree.cpp

```diff
@@ -35,7 +35,16 @@ GMSHPolygonTree::GMSHPolygonTree(GeoLib::Polygon* polygon, GMSHPolygonTree* pare
 {}
 
 GMSHPolygonTree::~GMSHPolygonTree()
-{}
+{
+	// the polylines are processed also by the childs, but the root is
+	// responsible to cleanup up
+	if (_parent == nullptr) { // root
```

Comment:
If `_parent` is a `nullptr` the node of the tree is the root. Only the root node is responsible for cleanup.